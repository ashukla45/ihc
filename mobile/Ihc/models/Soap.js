export default class Soap {
  // Insert any class methods here

  static extractFromForm(form, patientKey) {
    const soap = Object.assign({}, form);
    soap.patientKey = patientKey;
    soap.lastUpdated = new Date().getTime();
    return soap;
  }

  // Can pass in parameters to override defaults, mostly useful for tests
  static getInstance(patientKey = 'firstname&father&mother&20000101',
    date = '20180101', subjective = 'subjective', objective = 'objective',
    assessment = 'assessment', plan = 'plan', wishlist = 'wishlist', provider = 'doc',
    lastUpdated = new Date().getTime()) {
    return {
      patientKey, date, subjective, objective, assessment, plan, wishlist, provider, lastUpdated
    };
  }
}

Soap.schema = {
  name: 'Soap',
  properties: {
    patientKey: 'string',
    date: 'string',
    subjective: 'string?',
    objective: 'string?',
    assessment: 'string?',
    plan: 'string?',
    wishlist: 'string?',
    provider: 'string', // Doctor's name
    lastUpdated: 'int',
  }
};
