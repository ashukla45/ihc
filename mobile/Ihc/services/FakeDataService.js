/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function getPatient(patientKey) {
  return {};
}

export function getTriage(patientKey, date) {
  return {};
}

export function getSoap(patientKey, date) {
  return {};
}

export function getPatients(param) {
  return []; // Return whatever sample data you want
}

export function getUpdates(param) {
  return []; // Return whatever sample data you want
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
