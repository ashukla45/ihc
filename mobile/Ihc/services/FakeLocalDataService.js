/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function createPatient(patient) {
}

export function signinPatient(patientForm) {
}

export function updateStatus(patientKey, strDate, field, value) {
}

export function createDrugUpdate(update) {
}

export function updateSoap(update) {
}

export function getSoap(patientKey, strDate) {
}

export function updateTriage(update) {
}

export function getTriage(patientKey, strDate) {
}

export function getPatient(patientKey) {
}

export function getPatientsToUpload() {
}

export function lastSynced() {
}

export function handleDownloadedPatients(patients) {
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
