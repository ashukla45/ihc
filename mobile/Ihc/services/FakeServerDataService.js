/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests

export function createPatient(patient) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
      // rej(new Error("Fake error in FakeServerDataService"));
    }, 500);
  });
}

export function updateStatus(statusObj) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: put /patient/:key/drugUpdate
export function createDrugUpdate(update) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: put /patient/:key/soap/:date
export function updateSoap(update) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: get /patient/:key/soap/:date
// Returns SOAP object if it exists, or undefined if not
export function getSoap(patientKey, strDate) {
}

// Server endpoint: put /patient/:key/triage/:date
export function updateTriage(update) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: get /patient/:key/triage/:date
// Returns Triage object if it exists, or undefined if not
export function getTriage(patientKey, strDate) {
}

// Server endpoint: get /patient/:key
export function getPatient(patientKey) {
}

// Server endpoint: get /patient/:key/drugUpdates
export function getMedicationUpdates(patientKey) {
}

// Server endpoint: get /patients/statuses/:date
// Return the statuses of the patients that are active and for this date
export function getActiveStatuses() {
}

// Server endpoint: post /patients
export function updatePatients(patients) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: get /patients/:timestamp
export function getUpdatedPatients(lastSynced) {
}
/* eslint-enable */
