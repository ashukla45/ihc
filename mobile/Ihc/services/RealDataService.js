// This file should handle all the fetch() calls to the Express server
// and the local database calls to the Realm DB
// All lastUpdated fields on mobile-side should be handled within these
// functions

import Patient from '../models/Patient';
import Status from '../models/Status';
import Soap from '../models/Soap';
import Triage from '../models/Triage';
import DrugUpdate from '../models/DrugUpdate';
import Settings from '../models/Settings';

import {createPatientHelper} from './functions/CreatePatient';
import {signinPatientHelper} from './functions/SigninPatient';
import {updateStatusHelper} from './functions/UpdateStatus';
import {updateSoapHelper} from './functions/UpdateSoap';
import {createDrugUpdateHelper} from './functions/CreateDrugUpdate';
import {getSoapHelper} from './functions/GetSoap';
import {getTriageHelper} from './functions/GetTriage';
import {getPatientHelper} from './functions/GetPatient';
import {getPatientSelectRowsHelper} from './functions/GetPatientSelectRows';
import {updateTriageHelper} from './functions/UpdateTriage';
import {getMedicationUpdatesHelper} from './functions/GetMedicationUpdates';
import {uploadUpdatesHelper} from './functions/UploadUpdates';
import {downloadUpdatesHelper} from './functions/DownloadUpdates';

import Realm from 'realm';
import config from '../config.json';

// Must set the fetchUrl to the server's IP Address and Port
const fetchUrl = config.fetchUrl;

const realm = new Realm({
  schema: [Patient, Status, Soap, Triage, DrugUpdate, Settings],
  deleteRealmIfMigrationNeeded: true, // TODO: delete when done with dev
});

// Server endpoint: post /patient
export function createPatient(patient) {
  return createPatientHelper(patient, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/status/:date
// Check that patient exists locally, and if so then create a status object for them
export function signinPatient(patientForm) {
  return signinPatientHelper(patientForm, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/status/:date
// field: Any of fields from Status.schema
// value: should match the type that the Status.schema says
export function updateStatus(patientKey, strDate, field, value) {
  return updateStatusHelper(patientKey, strDate, field, value, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/drugUpdate
export function createDrugUpdate(update) {
  return createDrugUpdateHelper(update, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/soap/:date
export function updateSoap(update) {
  return updateSoapHelper(update, realm, fetchUrl);
}

// Server endpoint: get /patient/:key/soap/:date
// Returns SOAP object if it exists, or undefined if not
export function getSoap(patientKey, strDate) {
  return getSoapHelper(patientKey, strDate, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/triage/:date
export function updateTriage(update) {
  return updateTriageHelper(update, realm, fetchUrl);
}

// Server endpoint: get /patient/:key/triage/:date
// Returns Triage object if it exists, or undefined if not
export function getTriage(patientKey, strDate) {
  return getTriageHelper(patientKey, strDate, realm, fetchUrl);
}

// Server endpoint: get /patient/:key
export function getPatient(patientKey) {
  return getPatientHelper(patientKey, realm, fetchUrl);
}

// Server endpoint: get /patient/:key/drugUpdates
export function getMedicationUpdates(patientKey) {
  return getMedicationUpdatesHelper(patientKey, realm, fetchUrl);
}

// Server endpoint: get /patients/statuses
// Return the statuses of the patients that are active and for this date
export function getPatientSelectRows() {
  return getPatientSelectRowsHelper(realm, fetchUrl);
}

// Server endpoint: post /patients
export function uploadUpdates() {
  return uploadUpdatesHelper(realm, fetchUrl);
}

// Server endpoint: get /patients/:timestamp
export function downloadUpdates() {
  return downloadUpdatesHelper(realm, fetchUrl);
}
