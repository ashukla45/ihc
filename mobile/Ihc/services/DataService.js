/* global */
// Returns either fakeLocalData or RealData depending on config file
// Config file:
// "FakeDataServices": "true" or "false"
import * as fakeLocalData from '../services/FakeLocalDataService';
import * as localDataService from '../services/LocalDataService';

import * as fakeServerData from '../services/FakeServerDataService';
import * as serverDataService from '../services/ServerDataService';

import config from '../config.json';

// Dont use the Fakes in tests, so that we can test the LocalDataService itself,
// but should stub out its methods for the other tests
let localData = (config.fakeLocalDataService === 'true') ? fakeLocalData : localDataService;

// If we are testing the server data service, then use the real one
let serverData = (config.fakeServerDataService === 'true' && config.testingServerDataService == 'false')
  ? fakeServerData : serverDataService;

export {localData, serverData};
