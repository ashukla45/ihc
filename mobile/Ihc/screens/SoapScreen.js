import React, { Component } from 'react';
import {
  StyleSheet,
  Button,
  Text,
  View
} from 'react-native';
var t = require('tcomb-form-native');
var Form = t.form.Form;

import {localData} from '../services/DataService';
import Soap from '../models/Soap';
import {stringDate} from '../util/Date';
import Container from '../components/Container';

export default class SoapScreen extends Component<{}> {
  /*
   * Props:
   * name: patient's name for convenience
   * patientKey: string of patient's key
   * todayDate (optional, if doesn't exist, then assume date is for today,
   *   can be used for gathering old traige data from history)
   */
  constructor(props) {
    super(props);
    const todayDate = this.props.todayDate || stringDate(new Date());
    this.state = {
      formValues: {date: todayDate},
      errorMsg: null,
      todayDate: todayDate,
      successMsg: null,
      loading: false
    };
  }

  // TODO: Make form fields larger, more like textarea
  Soap = t.struct({
    date: t.String,
    subjective: t.maybe(t.String),
    objective: t.maybe(t.String),
    assessment: t.maybe(t.String),
    plan: t.maybe(t.String),
    wishlist: t.maybe(t.String),
    provider: t.String // Doctor's name
  });

  formOptions = {
    fields: {
      date: {
        editable: false,
      },
      subjective: {
        multiline: true,
      },
      objective: {
        multiline: true,
      },
      assessment: {
        multiline: true,
      },
      plan: {
        multiline: true,
      },
      wishlist: {
        multiline: true,
      },
    }
  }

  // Load existing SOAP info if it exists
  loadFormValues = () => {
    this.setState({ loading: true });
    const soap = localData.getSoap(this.props.patientKey, this.state.todayDate);
    if (!soap) {
      this.setState({loading: false});
      return;
    }

    this.setState({
      formValues: soap,
      loading: false
    });
  }

  componentDidMount() {
    this.loadFormValues();
  }

  submit = () => {
    if(!this.refs.form.validate().isValid()) {
      return;
    }
    this.setState({loading: true});
    const form = this.refs.form.getValue();
    const soap = Soap.extractFromForm(form, this.props.patientKey);

    try {
      localData.updateSoap(soap);
    } catch(e) {
      this.setState({errorMsg: e.message, successMsg: null});
      return;
    }

    // TODO: Submit on server too

    this.setState({
      successMsg: 'SOAP updated successfully',
      errorMsg: null,
      loading: false
    });
  }

  onFormChange = (value) => {
    this.setState({
      formValues: value,
    });
  }

  render() {
    return (
      <Container loading={this.state.loading} errorMsg={this.state.errorMsg} 
        successMsg={this.state.successMsg}>

        <Text style={styles.title}>
          Soap
        </Text>

        <View style={styles.form}>
          <Form ref="form"
            type={this.Soap}
            value={this.state.formValues}
            options={this.formOptions}
            onChange={this.onFormChange}
          />

          <Button onPress={this.submit}
            styles={styles.button}
            title="Update" />
        </View>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  form: {
    width: '80%',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});
