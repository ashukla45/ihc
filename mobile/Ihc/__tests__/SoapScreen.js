import 'react-native';
import React from 'react';
import SoapScreen from '../screens/SoapScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {localData} from '../services/DataService';
import sinon from 'sinon';
import Soap from '../models/Soap';

it('renders correctly', () => {
  sinon.stub(localData, 'getSoap').returns(Soap.getInstance());
  const json = renderer.create(
    <SoapScreen todayDate='20180101' />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
