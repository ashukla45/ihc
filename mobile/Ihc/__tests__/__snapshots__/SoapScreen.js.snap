// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Soap
      </Text>
      <View
        style={
          Object {
            "width": "80%",
          }
        }
      >
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                }
              }
            >
              Date
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Date"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={false}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={undefined}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "backgroundColor": "#eeeeee",
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#777777",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value="20180101"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                }
              }
            >
              Subjective (optional)
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Subjective (optional)"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={true}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value="subjective"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                }
              }
            >
              Objective (optional)
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Objective (optional)"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={true}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value="objective"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                }
              }
            >
              Assessment (optional)
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Assessment (optional)"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={true}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value="assessment"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                }
              }
            >
              Plan (optional)
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Plan (optional)"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={true}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value="plan"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                }
              }
            >
              Wishlist (optional)
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Wishlist (optional)"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={true}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value="wishlist"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                }
              }
            >
              Provider
            </Text>
            <View
              style={Object {}}
            >
              <TextInput
                accessibilityLabel="Provider"
                allowFontScaling={true}
                autoCapitalize={undefined}
                autoCorrect={undefined}
                autoFocus={undefined}
                blurOnSubmit={undefined}
                clearButtonMode={undefined}
                clearTextOnFocus={undefined}
                editable={undefined}
                enablesReturnKeyAutomatically={undefined}
                keyboardAppearance={undefined}
                keyboardType={undefined}
                maxLength={undefined}
                multiline={undefined}
                numberOfLines={undefined}
                onBlur={undefined}
                onChange={undefined}
                onChangeText={[Function]}
                onContentSizeChange={undefined}
                onEndEditing={undefined}
                onFocus={undefined}
                onKeyPress={undefined}
                onLayout={undefined}
                onSelectionChange={undefined}
                onSubmitEditing={undefined}
                placeholder={undefined}
                placeholderTextColor={undefined}
                returnKeyType={undefined}
                secureTextEntry={undefined}
                selectTextOnFocus={undefined}
                selectionColor={undefined}
                selectionState={undefined}
                style={
                  Object {
                    "borderColor": "#cccccc",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontSize": 17,
                    "height": 36,
                    "marginBottom": 5,
                    "paddingHorizontal": 7,
                    "paddingVertical": 7,
                  }
                }
                underlineColorAndroid="transparent"
                value="doc"
              />
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={undefined}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Update
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
